common/src/main/java/com/wynntils/features/inventory/CustomBankPagesFeature.java

```diff
@@ -0,0 +1,226 @@
+import com.wynntils.core.config.ConfigHolder;
+import com.wynntils.core.config.RegisterConfig;
+import com.wynntils.core.features.Feature;
+import com.wynntils.mc.event.ContainerClickEvent;
+import com.wynntils.mc.event.ContainerCloseEvent;
+import com.wynntils.mc.event.ScreenInitEvent;
+import com.wynntils.mc.event.SlotRenderEvent;
+import com.wynntils.utils.mc.McUtils;
+import com.wynntils.utils.render.RenderUtils;
+import com.wynntils.utils.render.Texture;
+import com.wynntils.utils.wynn.ContainerUtils;
+import java.util.List;
+import net.minecraft.ChatFormatting;
+import net.minecraft.client.gui.screens.inventory.AbstractContainerScreen;
+import net.minecraft.network.chat.Component;
+import net.minecraft.world.inventory.Slot;
+import net.minecraft.world.item.ItemStack;
+import net.minecraft.world.item.Items;
+import net.minecraftforge.eventbus.api.SubscribeEvent;
+import org.lwjgl.glfw.GLFW;
+
+@ConfigCategory(Category.INVENTORY)
+public class CustomBankPagesFeature extends Feature {
+    // Change to ranged integer when implemented
+    @RegisterConfig
+    public final Config<Integer> buttonOnePage = new Config<>(1);
+
+    @RegisterConfig
+    public final Config<Integer> buttonTwoPage = new Config<>(5);
+
+    @RegisterConfig
+    public final Config<Integer> buttonThreePage = new Config<>(9);
+
+    @RegisterConfig
+    public final Config<Integer> buttonFourPage = new Config<>(13);
+
+    @RegisterConfig
+    public final Config<Integer> buttonFivePage = new Config<>(17);
+
+    @RegisterConfig
+    public final Config<Integer> buttonSixPage = new Config<>(21);
+
+    private static final int MAX_BANK_PAGES = 21;
+    private static final int NEXT_PAGE_SLOT = 8;
+    private static final int PREVIOUS_PAGE_SLOT = 17;
+    private static final List<Integer> BUTTON_SLOTS = List.of(7, 16, 25, 34, 43, 52);
+    private static final List<Integer> QUICK_JUMP_DESTINATIONS = List.of(1, 5, 9, 13, 17, 21);
+
+    private boolean isBankScreen = false;
+    private boolean onLastPage = false;
+    private boolean quickJumping = false;
+    private int containerId;
+    private int currentPage = 1;
+    private int pageDestination = 1;
+    private List<Integer> customJumpDestinations;
+
+    @SubscribeEvent
+    public void onScreenInit(ScreenInitEvent e) {
+        if (!(e.getScreen() instanceof AbstractContainerScreen<?> screen)) return;
+        if (!Models.Container.isBankScreen(screen)) return;
+
+        isBankScreen = true;
+
+        containerId = screen.getMenu().containerId;
+        currentPage = Models.Container.getCurrentBankPage(screen);
+
+        customJumpDestinations = List.of(
+                buttonOnePage.get(),
+                buttonTwoPage.get(),
+                buttonThreePage.get(),
+                buttonFourPage.get(),
+                buttonFivePage.get(),
+                buttonSixPage.get());
+
+        if (onLastPage) {
+            quickJumping = false;
+            pageDestination = currentPage;
```

Comment:
What is pageDestination representing? I don't follow the flow here.

Reply:
pageDestination is the page you want to jump to, so starts out as 1 so when you open the bank it doesn't auto start jumping somewhere.

When you click the next/previous buttons that will increment/decrement the value so that it doesn't auto jump back to the page you were on. This is also needed because the scroll feature can't just tell this feature that a scroll was performed.

Then once you click one of the jump it will use the destination to find which of the default destinations is the closest and each time a new page is rendered, if you're on the last page you own, it will set this page to your destination to stop it from trying to go back to the destination if you hit the previous page button.

Since I don't own page 18, if I tried to jump to 18 it would take me to 17 and then if I clicked the previous button it would try and go back go 18 so destination has to get set to current page if on the last page

Reply:
But if you were at the last page and wanted to quickjump to the first page? Wouldn't this cancel that behavior?

Reply:
Looking at it, yes I think it would.

Adding `&& pageDestination > currentPage` should fix it I think